src/config-import.js

```diff
@@ -215,6 +247,164 @@ async function findPresetsFile(entries) {
   return { presets, fields }
 }
 
+/**
+ * @param {ReadonlyArray<Entry>} entries
+ * @returns {Promise<TranslationsFile | undefined>}
+ */
+async function findTranslationsFile(entries) {
+  const translationEntry = entries.find(
+    (entry) => entry.filename === 'translations.json'
+  )
+  if (!translationEntry) return
+
+  /** @type {unknown} */
+  let result
+  try {
+    result = await json(await translationEntry.openReadStream())
+  } catch (err) {
+    throw new Error('Could not parse translations.json')
+  }
+  assert(isRecord(result), 'Invalid translations.json file')
+  return result
+}
+
+/**
+ * @param {string} lang
+ * @param {Record<string, unknown>} languageTranslations
+ *
+ */
+function* translationsForLanguage(lang, languageTranslations) {
+  const { language: languageCode, region: regionCode } = parseBCP47(lang)
+  if (!languageCode) {
+    warnings.push(new Error(`invalid translation language ${lang}`))
+    return
+  }
+  for (const [schemaNameRef, languageTranslationsForDocType] of Object.entries(
+    languageTranslations
+  )) {
+    // TODO: push to warnings when removing categories from default config
```

Comment:
What does this mean? And should we actually add to `warnings` in this case, uncommenting line 287?